core/src/main/scala/org/apache/gearpump/metrics/Metrics.scala

```diff
@@ -48,6 +51,15 @@ object Metrics extends ExtensionId[Metrics] with ExtensionIdProvider {
   val LOG: Logger = LogUtil.getLogger(getClass)
   import org.apache.gearpump.util.Constants._
 
+  sealed trait MetricType
+  case class Histogram(count: Long, min: Long, max: Long, mean: Double, stddev: Double, median: Double, p75: Double, p95: Double, p98: Double, p99: Double, p999: Double) extends MetricType
+  case class Counter(name: String, value: Long) extends MetricType
+  case class Meter(name: String, count: Long, mean_rate: Double, m1: Double, m5: Double, m15: Double, rate_unit: String) extends MetricType
+  case class Timer(name: String, count: Long, min: Double, max: Double, mean: Double, stddev: Double,
+                   median: Double, p75: Double, p95: Double, p98: Double, p99: Double, p999: Double, mean_rate: Double,
+                   m1: Double, m5: Double, m15: Double, rate_unit: String, duration_unit: String) extends MetricType
```

Comment:
Most APIs uses camelcase for parameters, but rate_unit and duration_unit here are not.